src/main/scala/diplomacy/Nodes.scala

```diff
@@ -223,31 +645,83 @@ trait InwardNode[DI, UI, BI <: Data] extends BaseNode
     accPI += ((index, node, binding, p, sourceInfo))
   }
 
-  protected[diplomacy] lazy val iBindings = { iRealized = true; accPI.result() }
+  /** Ends the binding accumulation stage and returns all the input bindings to this node.
+    *
+    * Evaluating this lazy val will mark the inwards bindings as frozen,
+    * preventing subsequent bindings from being created via [[iPush]].
+    *
+    * The bindings are each a tuple of:
+    * - numeric index of this binding in the other end of [[OutwardNode]].
+    * - [[OutwardNode]] on the other end of this binding.
+    * - [[NodeBinding]] describing the type of binding.
+    * - a [[Parameters]] instance.
+    * - [[SourceInfo]] for source-level error reporting.
+    */
+  protected[diplomacy] lazy val iBindings: immutable.Seq[(Int, OutwardNode[DI, UI, BI], NodeBinding, Parameters, SourceInfo)] = { iRealized = true; accPI.result() }
 
+  /** resolved STAR binding of inward nodes: how many connections the star represents. */
   protected[diplomacy] val iStar: Int
+  /** A mapping to convert Node binding index to port range.
+    *
+    * index: the index of connected [[OutwardNode]]
+    * element: port range of connected [[OutwardNode]]
+    */
   protected[diplomacy] val iPortMapping: Seq[(Int, Int)]
+
+  /** "Forward" an input connection through this node so that the node can be removed from the graph.
+    *
+    *  @return None if no forwarding is needing.
+    */
   protected[diplomacy] def iForward(x: Int): Option[(Int, InwardNode[DI, UI, BI])] = None
-  protected[diplomacy] val diParams: Seq[DI] // from connected nodes
-  protected[diplomacy] val uiParams: Seq[UI] // from this node
+  /** Downstream inward parameters.
+    *
+    * Generated from the other nodes this node is connected to.
+    */
+  protected[diplomacy] val diParams: Seq[DI]
+  /** Upstream inward parameters.
+    *
+    * Generated from this node.
```

Comment:
Generated by this node and sent to the nodes connected to the inward side of this node.